### DISCUSSION TITLE
Training with Colab fails

### DISCUSSION THREAD

**Discussion by KruemelmonsterK** [talmolab/sleap#1381] [Help!]
When I use our Notebooks on Colab (Notebook on my own data or Notebook on example data), I get the following error message for training:

....
2023-07-10 15:23:33.937251: W tensorflow/core/grappler/costs/op_level_cost_estimator.cc:690] Error in PredictCost() for the op: op: "CropAndResize" attr { key: "T" value { type: DT_FLOAT } } attr { key: "extrapolation_value" value { f: 0 } } attr { key: "method" value { s: "bilinear" } } inputs { dtype: DT_FLOAT shape { dim { size: -34 } dim { size: -35 } dim { size: -36 } dim { size: 1 } } } inputs { dtype: DT_FLOAT shape { dim { size: -2 } dim { size: 4 } } } inputs { dtype: DT_INT32 shape { dim { size: -2 } } } inputs { dtype: DT_INT32 shape { dim { size: 2 } } } device { type: "CPU" vendor: "GenuineIntel" model: "111" frequency: 2199 num_cores: 2 environment { key: "cpu_instruction_set" value: "AVX SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2" } environment { key: "eigen" value: "3.4.90" } l1_cache_size: 32768 l2_cache_size: 262144 l3_cache_size: 57671680 memory_size: 268435456 } outputs { dtype: DT_FLOAT shape { dim { size: -2 } dim { size: -37 } dim { size: -38 } dim { size: 1 } } }
2023-07-10 15:23:36.213904: W tensorflow/core/grappler/costs/op_level_cost_estimator.cc:690] Error in PredictCost() for the op: op: "CropAndResize" attr { key: "T" value { type: DT_FLOAT } } attr { key: "extrapolation_value" value { f: 0 } } attr { key: "method" value { s: "bilinear" } } inputs { dtype: DT_FLOAT shape { dim { size: -34 } dim { size: -35 } dim { size: -36 } dim { size: 1 } } } inputs { dtype: DT_FLOAT shape { dim { size: -2 } dim { size: 4 } } } inputs { dtype: DT_INT32 shape { dim { size: -2 } } } inputs { dtype: DT_INT32 shape { dim { size: 2 } } } device { type: "CPU" vendor: "GenuineIntel" model: "111" frequency: 2199 num_cores: 2 environment { key: "cpu_instruction_set" value: "AVX SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2" } environment { key: "eigen" value: "3.4.90" } l1_cache_size: 32768 l2_cache_size: 262144 l3_cache_size: 57671680 memory_size: 268435456 } outputs { dtype: DT_FLOAT shape { dim { size: -2 } dim { size: -37 } dim { size: -38 } dim { size: 1 } } }
Unable to use Qt backend for matplotlib. This probably means Qt is running headless.
Unable to use Qt backend for matplotlib. This probably means Qt is running headless.
INFO:sleap.nn.training:Finished trainer set up. [9.1s]
INFO:sleap.nn.training:Creating tf.data.Datasets for training data generation...
INFO:sleap.nn.training:Finished creating training datasets. [51.0s]
INFO:sleap.nn.training:Starting training loop...
Epoch 1/200
2023-07-10 15:24:32.848600: W tensorflow/core/framework/cpu_allocator_impl.cc:82] Allocation of 100663296 exceeds 10% of free system memory.
2023-07-10 15:24:33.946254: W tensorflow/core/framework/cpu_allocator_impl.cc:82] Allocation of 100663296 exceeds 10% of free system memory.
2023-07-10 15:24:33.946401: W tensorflow/core/framework/cpu_allocator_impl.cc:82] Allocation of 226492416 exceeds 10% of free system memory.
2023-07-10 15:24:34.233062: W tensorflow/core/framework/cpu_allocator_impl.cc:82] Allocation of 226492416 exceeds 10% of free system memory.
2023-07-10 15:24:35.312030: W tensorflow/core/framework/cpu_allocator_impl.cc:82] Allocation of 113246208 exceeds 10% of free system memory.

**Comment by roomrys** [comment]
Hi @KruemelmonsterKn

It looks like you are running into some memory issues. Some things to try:
1. If you are tracking larger features (greater that 4 pixels), then you can try setting the input scaling to 0.5. 
2. If you are using a bottom-up or single animal model, try using top-down which is SLEAP's most memory efficient model.

Let us know if you run into any trouble.

Thanks,
Liezl
